porespy/networks/_AI_diffusive_size_factor.py

```diff
@@ -0,0 +1,405 @@
+    return diff_size_factor
+
+
+def _calc_bound_box_bi(regions, pore_1, pore_2):
+    '''
+
+    Parameters
+    ----------
+    regions : ndarray
+        3D image of the segmented regions from which the bounding box of
+        a local pore to pore region will be extracted.
+    pore_1 : scalar
+        Label of pore region1 in a pair of pore to pore connection (throat)
+    pore_2 : scalar
+        Label of pore region1 in a pair of pore to pore connection (throat)
+
+    Returns
+    -------
+    b_box : array
+        An array containing the coordinates of the bounding box that covers
+        region pore_1 and region pore_2. The orders of coordinates are:
+        x_min, y_min, z_min, x_max, y_max, z_max.
+
+    '''
+    slice_x, slice_y, slice_z = spim.find_objects(regions == pore_1+1)[0]
+    slice_x2, slice_y2, slice_z2 = spim.find_objects(regions == pore_2+1)[0]
+    min_box = [min(slice_x.start, slice_x2.start),
+               min(slice_y.start, slice_y2.start),
+               min(slice_z.start, slice_z2.start)]
+    max_box = [max(slice_x.stop, slice_x2.stop), max(slice_y.stop, slice_y2.stop),
+               max(slice_z.stop, slice_z2.stop)]
+    b_box = np.hstack([min_box, max_box])
+    return b_box
+
+
+def _resize_to_AI_input(im):
+    '''
+    Parameters
+    ----------
+    im : ndarray
+        3D image of a pair of pore to pore regions (conduit) cropped from segmented image of 
+        a porous medium.
+
+    Returns
+    -------
+    result : dict
+        Contains Resized image of the cropped regions and its zoom ratio. If original image(im)
+        is of shape (n,n,n) resizing includes one step of zoom to a (64,64,64) image. Otherwise,
+        the original image will be first zero padded to its maximum dimension (nmax,nmax,nmax) before
+        zoom step.
+
+    '''
+    if len(np.unique(im.shape)) != 1:
+        x, y, z = np.shape(im)
+        max_size = max([x, y, z])
+        x_diff, y_diff, z_diff = max_size-np.array([x, y, z])
+        XB = int(np.round(x_diff/2))  # padding before axis x
+        XA = int(x_diff-XB)  # padding after axis x
+        YB = int(np.round(y_diff/2))
+        YA = int(y_diff-YB)
+        ZB = int(np.round(z_diff/2))
+        ZA = int(z_diff-ZB)
+        im = np.pad(im, ((XB, XA), (YB, YA), (ZB, ZA)), 'constant',
+                    constant_values=0)
+    zm_ratio = 64/im.shape[0]
+    resized_im = zm(im, zoom=[zm_ratio, zm_ratio, zm_ratio], order=0)
+    result = {'zm_ratio': zm_ratio, 'resized_im': resized_im}
+    return result
+
+
+def find_conns(im):
+    '''
+    Parameters
+    ----------
+    im : ndarray
+        A segmented image of a porous medium.
+
+    Returns
+    -------
+    t_conns : array
+        An Nt by 2 addat containing the throats' connections in the segmented image.
+
+    '''
+    struc_elem = ball
+    slices = spim.find_objects(im)
+    Ps = np.arange(1, np.amax(im)+1)
+    t_conns = []
+    for i in tqdm(Ps, file=sys.stdout):
```

Comment:
This line should be ``tqdm(Ps, **settings.tqdm)``. assuming you've called ``from porespy import settings``